src/dual.rs

```diff
@@ -0,0 +1,12 @@
+//! The dual value measures the increase in the objective function's value per unit
+//! increase in the variable's value. The dual value for a constraint is nonzero only when
+//! the constraint is equal to its bound. Also known as the shadow price
+//!
+
+/// A problem dual values (shadow prices)
+pub trait DualValues {
+    /// Get all dual values for a given problem
+    fn get_dual_values(&self) -> &[f64];
+    /// Get the dual value of a problem by index
+    fn get_dual_value(&self, constraint_index: usize) -> f64;
+}
```

Comment:
I don't think the trait should work like that. This forces either the solver's internal representation to always have the dual values as a continuous slice of doubles, or us to always compute the dual values when we don't need them. 

I think the trait should allow code to look like the following : 

```rust
let my_constraint : ConstraintReference = pb.add_constraint( x + y << 7)
let solution = pb.solve()?
let constraint_dual =  solution.get_dual().value(my_constraint)
```

This means : 
  - creating `add_constraint(&mut self, Constraint) -> ConstraintReference`
  - creating a `SolutionWithDual` which allows computing the dual values, and a `DualValues` trait (or just a struct) that takes a  `ConstraintReference` and returns the corresponding dual value.